### PR TITLE
[Optimalcontrol] Use finer sparsity

### PR DESCRIPTION
**Pull request by S-Dafarra**
YAPR (yet another PR) on ``optimalcontrol``:
- Implements interfaces to take into account the sparsity of constraints and dynamical systems.
- Some bugfix

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,7 +25,7 @@ set(VARS_PREFIX "iDynTree")
 
 set(${VARS_PREFIX}_MAJOR_VERSION 0)
 set(${VARS_PREFIX}_MINOR_VERSION 11)
-set(${VARS_PREFIX}_PATCH_VERSION 1)
+set(${VARS_PREFIX}_PATCH_VERSION 2)
```
Comment (traversaro) [inline CMakeLists.txt]: I am afraid I did not document this, but I think we should follow the YARP versioning scheme documented in https://github.com/robotology/yarp/blob/master/.github/CONTRIBUTING.md#example , so the devel version should have thier patch version that should start from a big number such as `100`. I should have done this after the 0.11 release, but as you may imagine not being an automatic step I forgot to do it. : )

File: src/optimalcontrol/tests/MultipleShootingTest.cpp [hunk 2]
```diff
@@ -77,6 +77,23 @@ class TestSystem : public iDynTree::optimalcontrol::DynamicalSystem {
         dynamicsDerivative(1, 2) = 2.0;
         return true;
     }
+
+    virtual bool dynamicsStateFirstDerivativeSparsity(iDynTree::optimalcontrol::SparsityStructure& stateSparsity) override{
```
Comment (traversaro) [inline src/optimalcontrol/tests/MultipleShootingTest.cpp]:
```suggestion
    virtual bool dynamicsStateFirstDerivativeSparsity(iDynTree::optimalcontrol::SparsityStructure& stateSparsity) override {
```

File: src/optimalcontrol/tests/OCProblemTest.cpp [hunk 3]
```diff
@@ -73,6 +73,23 @@ class TestSystem : public iDynTree::optimalcontrol::DynamicalSystem {
         dynamicsDerivative(1, 2) = 2.0;
         return true;
     }
+
+    virtual bool dynamicsStateFirstDerivativeSparsity(iDynTree::optimalcontrol::SparsityStructure& stateSparsity) override{
```
Comment (traversaro) [inline src/optimalcontrol/tests/OCProblemTest.cpp]:
```suggestion
    virtual bool dynamicsStateFirstDerivativeSparsity(iDynTree::optimalcontrol::SparsityStructure& stateSparsity) override {
```

File: CMakeLists.txt (hunk 1 above)
Comment (S-Dafarra) [reply CMakeLists.txt]: Is ``102`` ok?

Comment (traversaro) [reply CMakeLists.txt]: ok!

Comment (S-Dafarra) [reply CMakeLists.txt]: Done in https://github.com/robotology/idyntree/pull/517/commits/27c1008229f18d1adba528ca4199f9f3cffb6122

**Comment by traversaro** [comment]
For me it is ok.

**Comment by S-Dafarra** [comment]
Thanks!

**Pull request merged**